### ISSUE TITLE
RESOLVED: `no-unscoped-css-nesting-selector` incorrectly reports nested selector with an ancestor style rule

### ISSUE THREAD

**Issue by ShayanTheNerd** [sindresorhus/eslint-plugin-unicorn#3706]
<ins>
  <b>This was caused by a syntax error from somewhere else in the code, so it was a mistake on my end, sorry.</b>
</ins>

<br /><br />

<del>
**The environment**
ESLint version: 10.10.0
@eslint/css version: 2.0.0
eslint-plugin-unicorn version: 75.0.0

<br />

**The setup**
Using [unicorn/no-unscoped-css-nesting-selector](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v75.0.0/docs/rules/no-unscoped-css-nesting-selector.md) with this configuration:
```js
{
  'unicorn/no-unscoped-css-nesting-selector': (
    isEnabled(tailwind) ? ['error', { scopingRootAtRules: ['utility', 'custom-variant'] }] : 'error'
  ),
}
```
and with this code:
```css
@layer base {
  input {
    caret-color: var(--color-brand);

    &[type='search'] {
      appearance: textfield;
    }
  }
}
```

<br />

**Actual behavior**

The `&` selector is reported:
>  Do not use a CSS nesting selector unless an ancestor scoping context applies. eslint(unicorn/no-unscoped-css-nesting-selector)

<br />

**Expected behavior**
No linting issue should be reported because the nesting selector has an ancestor style rule (`input`) that provides the scoping context.
</del>
